app/models/student.rb

```diff
@@ -33,18 +33,42 @@ class Student < ApplicationRecord
   scope :ine_not_found, -> { where(ine_not_found: true) }
   scope :with_ine, -> { where(ine_not_found: false) }
   scope :with_biological_sex, -> { where(biological_sex: %i[male female]) }
+  scope :with_known_postal_code, -> { where.not(address_postal_code: nil) }
+
   scope :with_known_birthplace, -> { where.not(birthplace_country_insee_code: nil) }
   scope :with_valid_birthplace, lambda {
     with_known_birthplace
+      .with_valid_birthplace_city
       .where.not(birthplace_country_insee_code: InseeCountryCodeMapper::REJECTED_CODES.keys)
   }
 
+  scope :with_known_address, -> { where.not(address_country_code: nil) }
+  scope :with_valid_address, lambda {
+    with_known_address
+      .with_known_postal_code
+      .with_valid_address_city
+      .where.not(address_country_code: InseeCountryCodeMapper::REJECTED_CODES.keys)
+  }
+
+  scope :with_valid_address_city, lambda {
+    where.not(
+      "students.address_country_code IN (?) AND students.address_city_insee_code IS NULL",
+      %w[100 99100]
```

Comment:
donc en fait c'est `lives_in_france`?

Reply:
non

Reply:
si l'adress country code doit être dans 100 99100 ?

Reply:
nope

Reply:
c'est lives in france + pas de code city null ?

Reply:
Y'a un .not devant. On avait déjà discuté de ce filtre plusieurs fois, il est bon.

Il provient de là : https://github.com/betagouv/aplypro/commit/f7b5c84bd81cbc0e29e366ca1e0b76ace58f5ee4

Reply:
On check "la présence de la ville si le pays est france"

Reply:
et en quoi tu peux pas faire `lives_in_france.with_valid_city_code` ? on en discute encore et toujours parce que c'est obscure comme syntaxe de NOT un valid state

Reply:
ah c'est pour les étrangers maybe

Reply:
Par ce que je veux pas un filtre qui me garde que des gens qui vivent en france. Je veux un filtre que me garde que des gens qui SOIT on france = city, soit ont un pays étranger. Et c'est ce que fait ma requête.

Si t'as une meilleure syntaxe je prend, pour moi c'est la clair à lire. on ne veux pas de français sans ville. 
`"not ( in france AND city empty)"`